json-cache/service-worker.js

```diff
@@ -0,0 +1,53 @@
+// [Working example](/serviceworker-cookbook/json-cache/).
+
+var CACHE_NAME = 'dependencies-cache';
+
+self.addEventListener('install', function(event) {
+  // Perform the install step:
+  // * Load a JSON file from server
+  // * Parse as JSON
+  // * Add files to the cache list
+
+  event.waitUntil(
+    caches.open(CACHE_NAME)
+      .then(function(cache) {
+        // With the cache opened, load a JSON file containing an array of files to be cached
+        return fetch('files-to-cache.json').then(function(response) {
+          // Once the contents are loaded, convert the raw text to a JavaScript object
+          return response.json();
+        }).then(function(files) {
+          // Use cache.addAll just as you would a hardcoded array of items
+          console.log('[install] Adding files from JSON file: ', files);
+          return cache.addAll(files);
+        });
+      })
+      .then(function() {
+        // Message to simply show the lifecycle flow
+        console.log('[install] All required resources have been cached, we\'re good!');
+      })
+  );
+});
+
+self.addEventListener('fetch', function(event) {
+  event.respondWith(
+    caches.match(event.request)
+      .then(function(response) {
+        // Cache hit - return the response from the cached version
+        if (response) {
+          console.log('[fetch] Returning from cache: ', event.request.url);
+          return response;
+        }
+
+        // Not in cache - return the result from the live server
+        // `fetch` is essentially a "fallback"
+        console.log('[fetch] Returning from server: ', event.request.url);
+        return fetch(event.request);
+      }
+    )
+  );
+});
+
+self.addEventListener('activate', function() {
+  // Message to simply show the lifecycle flow
+  console.log('[activate] Activating service worker!');
```

Comment:
Forgot to mention, but you should to call [calim()](https://developer.mozilla.org/en-US/docs/Web/API/Clients/claim) here.